File: InfoInspector/MissingAlbumArtTransformer.m
//
//  MissingAlbumArtTransformer.m
//  Cog
//
//  Created by Vincent Spader on 3/8/09.
//  Copyright 2009 __MyCompanyName__. All rights reserved.
//

#import "MissingAlbumArtTransformer.h"


@implementation MissingAlbumArtTransformer

+ (Class)transformedValueClass { return [NSImage class]; }
+ (BOOL)allowsReverseTransformation { return NO; }

// Convert from NSImage to NSImage
- (id)transformedValue:(id)value {
    if (value == nil) {
		return [NSImage imageNamed:@"missingArt"];
	}
	
	return value;
}

@end

File: InfoInspector/MissingAlbumArtTransformer.swift
import Foundation
import AppKit

@objc(MissingAlbumArtTransformer)
class MissingAlbumArtTransformer: ValueTransformer {
    override class func transformedValueClass() -> AnyClass {
        return NSImage.self
    }

    override class func allowsReverseTransformation() -> Bool {
        return false
    }

    // Convert from NSImage to NSImage
    override func transformedValue(_ value: Any?) -> Any? {
        if value == nil {
            return NSImage(named: "missingArt")
        }

        return value
    }
}
